standalone/src/main/java/com/criteo/publisher/samples/standalone/NativeRecyclerViewActivity.java

```diff
@@ -0,0 +1,168 @@
+package com.criteo.publisher.samples.standalone;
+
+import static com.criteo.publisher.samples.standalone.CriteoSampleApplication.CRITEO_NATIVE_AD_UNIT;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import android.os.Bundle;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.recyclerview.widget.RecyclerView.ViewHolder;
+import com.criteo.publisher.CriteoErrorCode;
+import com.criteo.publisher.advancednative.CriteoMediaView;
+import com.criteo.publisher.advancednative.CriteoNativeAd;
+import com.criteo.publisher.advancednative.CriteoNativeAdListener;
+import com.criteo.publisher.advancednative.CriteoNativeLoader;
+import com.criteo.publisher.advancednative.CriteoNativeRenderer;
+import com.criteo.publisher.advancednative.RendererHelper;
+import com.squareup.picasso.Picasso;
+import java.util.ArrayList;
+import java.util.List;
+
+public class NativeRecyclerViewActivity extends AppCompatActivity {
+
+  private MyAdapter adapter;
+  private CriteoNativeLoader nativeLoader;
+
+  private int dummyContentCount = 12;
+  private int nativeAdPosition = 3; // zero-indexed position
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_native_recycler_view);
+
+    adapter = new MyAdapter();
+    RecyclerView recyclerView = findViewById(R.id.recyclerView);
+    recyclerView.setLayoutManager(new LinearLayoutManager(this));
+    recyclerView.setAdapter(adapter);
+
+    // add some dummy content
+    for (int i=0; i<dummyContentCount; i++) {
```

Comment:
Are you using any formatter? I recommend using Criteo's Java code style (will send you the link).

What triggered this question is the lack of spacing around the `<` operator.

Reply:
I use Criteo formatter, but I didn't run them all the time. Have reformatted the code, thanks for reminding!